cmds/exp/tcpdump/gen.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build ignore
+
+package main
+
+import (
+	"bufio"
+	"io/ioutil"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func main() {
+	file, err := os.Open("/etc/services")
+	if err != nil {
+		log.Fatalf("Error opening /etc/services: %v", err)
+	}
+	defer file.Close()
+
+	re := regexp.MustCompile(`^(\S+)\s+(\d+)/\S+`)
+
+	ports := make(map[string]string)
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") || len(strings.TrimSpace(line)) == 0 {
+			continue
+		}
+		matches := re.FindStringSubmatch(line)
+		if len(matches) == 3 {
+			service := matches[1]
+			port := matches[2]
+			ports[port] = service
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		log.Fatalf("Error reading /etc/services: %v", err)
+	}
+
+	// Generate the content for the Go file
+	content := `// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+var wellKnownPortsMap = map[string]string{
+`
+
+	// Append the ports to the content
+	for port, service := range ports {
+		content += `    "` + port + `": "` + service + `",` + "\n"
+	}
+	content += "}\n"
+
+	// Write the content to the Go file
+	if err := ioutil.WriteFile("well_known_ports.go", []byte(content), 0644); err != nil {
```

Comment:
one little thing I do in this case: 
you can use exec.Command to start gofmt, and get the formatted content to write to the file.

Reply:
and note, if you do that, you can write unformatted go code, and let gofmt take the job of making it look good. Your print statement is then very simple

Reply:
This is a very good idea, i added the exec of gofmt.